src/Controllers/AdminController.php

```
@@ -778,6 +801,117 @@ public function billing(): void
+    {
+        $includedCredits = max(0, (int) ($packageSnapshot['included_ai_credits'] ?? 0));
+        [$periodStart, $periodEnd, $now] = $this->currentBillingPeriod($subscription);
+
+        if (!$this->db->tableExists('ai_usage_events')) {
+            return $this->emptyAiUsageSummary($includedCredits, $periodStart, $periodEnd, false);
+        }
+
+        try {
+            $service = ($this->aiUsageSummaryFactory)($this->db);
+            if (!$service instanceof AiUsageSummaryService) {
+                throw new \RuntimeException('Invalid AI usage summary service');
```

PHPMD (2.15.0)

[error] Missing class import via use statement (line '821', column '27'). (undefined)

(MissingImport)